cli/crates/federated-dev/src/dev/websockets.rs

```diff
@@ -31,16 +34,32 @@ impl WebsocketAccepter {
     }
 
     pub async fn handler(mut self) {
-        while let Some(connection) = self.sockets.recv().await {
+        while let Some(mut connection) = self.sockets.recv().await {
             let gateway = self.gateway.clone();
 
             tokio::spawn(async move {
-                let Some((websocket, session)) = accept_websocket(connection, &gateway).await else {
-                    log::warn!("Failed to accept websocket connection");
-                    return;
-                };
-
-                websocket_loop(websocket, session).await;
+                let accept_future = tokio::time::timeout(
+                    CONNECTION_INIT_WAIT_TIMEOUT,
+                    accept_websocket(&mut connection, &gateway),
+                );
+
+                match accept_future.await {
+                    Ok(Some(session)) => websocket_loop(connection, session).await,
+                    Ok(None) => {
+                        log::warn!("Failed to accept websocket connection");
+                    }
+                    Err(_) => {
+                        log::info!("Connection wasn't initialised on time, dropping");
+                        connection
+                            .send(
+                                Message::close(4408, "Connection initialisation timeout")
```

Comment:
I haven't dug in the api so maybe it's obvious, but what is 4408 here? (a port?)